src/lightning/fabric/utilities/spike.py

```diff
@@ -0,0 +1,165 @@
+import json
+import operator
+import os
+import warnings
+from typing import Any, Dict, List, Literal, Optional, type_checking, Union
+
+import torch
+from lightning_utilities.core.imports import compare_version
+
+_TORCHMETRICS_GREATER_EQUAL_1_0_0 = compare_version("torchmetrics", operator.ge, "1.0.0")
+
+if type_checking:
+    from lightning.fabric.fabric import Fabric
+
+
+class SpikeDetection:
+    """Spike Detection Callback.
+
+    Terminates training with a ``TrainingSpikeException`` when a loss-spike was detected and
+    saves the batches to skip when resuming to a file.
+
+    Currently we skip the current and the previous batch since it is unclear, whether the previous batch
+    altered the weights in a way that it causes the spike or just the current batch is corrupted somehow.
```

Comment:
```suggestion
    We skip the current and the previous batch since it is unclear whether the previous batch
    altered the weights in a way that it causes the spike or just the current batch is corrupted somehow.
```
Not sure about the comma, but suggest to remove the "Currently" in the beginning to avoid frequent word repetition